kybra/compiler/kybra_post_install/src/main.rs

```diff
@@ -0,0 +1,402 @@
+    let python_stdlib_modules = rustpython_vm::py_freeze!(dir = "src/Lib");
+
+    #[cfg(all(not(python_stdlib_exists), ic_build))]
+    let python_stdlib_bytecode = python_stdlib_modules.bytes.to_vec();
+
+    #[cfg(all(not(python_stdlib_exists), ic_build))]
+    std::fs::write(python_stdlib_path, &python_stdlib_bytecode).map_err(|e| error_to_string(&e))?;
+
+    #[cfg(all(python_stdlib_exists, ic_build))]
+    let python_stdlib_bytecode =
+        std::fs::read(python_stdlib_path).map_err(|e| error_to_string(&e))?;
+
+    #[cfg(not(ic_build))]
+    let python_stdlib_bytecode = vec![];
+
+    Ok(python_stdlib_bytecode)
+}
+
+fn add_permissions(canister_name: &str, dfx_network: &str) -> Result<(String, bool), String> {
+    let canister_id = get_canister_id(canister_name, dfx_network)?;
+    let canister_already_its_own_controller =
+        get_canister_already_its_own_controller(canister_name, dfx_network, &canister_id)?;
+
+    if !canister_already_its_own_controller {
+        add_controller(canister_name, dfx_network, &canister_id)?;
+    }
+
+    Ok((canister_id, canister_already_its_own_controller))
+}
+
+fn get_canister_id(canister_name: &str, dfx_network: &str) -> Result<String, String> {
+    let canister_id_output = Command::new("dfx")
+        .arg("canister")
+        .arg("id")
+        .arg("--network")
+        .arg(dfx_network)
+        .arg(canister_name)
+        .output()
+        .map_err(|_| create_error_string("Failed to execute the dfx canister id command"))?;
+
+    let canister_id = String::from_utf8_lossy(&canister_id_output.stdout)
+        .trim()
+        .to_string();
+
+    Ok(canister_id)
+}
+
+fn get_canister_already_its_own_controller(
+    canister_name: &str,
+    dfx_network: &str,
+    canister_id: &str,
+) -> Result<bool, String> {
+    let current_controllers_output = Command::new("dfx")
+        .arg("canister")
+        .arg("info")
+        .arg("--network")
+        .arg(dfx_network)
+        .arg(canister_name)
+        .output()
+        .map_err(|_| create_error_string("Failed to execute the dfx canister info command"))?;
+
+    if !current_controllers_output.status.success() {
+        return Err(create_error_string(&String::from_utf8_lossy(
+            &current_controllers_output.stderr,
+        )));
+    }
+
+    let canister_already_its_own_controller =
+        String::from_utf8_lossy(&current_controllers_output.stdout)
+            .to_string()
+            .contains(&canister_id);
+
+    Ok(canister_already_its_own_controller)
+}
+
+fn add_controller(canister_name: &str, dfx_network: &str, canister_id: &str) -> Result<(), String> {
+    let add_controller_output = Command::new("dfx")
+        .arg("canister")
+        .arg("update-settings")
+        .arg("--network")
+        .arg(dfx_network)
+        .arg("--add-controller")
+        .arg(&canister_id)
+        .arg(canister_name)
+        .output()
+        .map_err(|_| create_error_string("Failed to execute the dfx command"))?;
```

Comment:
I'm seeing a lot of duplication in these dfx calls. I wonder if we can abstract it away...